### PR TITLE
Validation zhuzh-up: isolated nodes

### PR DESCRIPTION
**Pull request by KasiaKoz**
Resolves: https://github.com/arup-group/genet/issues/123 by adding methods to:
- check if there are any isolated nodes in the graph
- list their IDs
- remove them

with ample logging. 

Adds info on isolated nodes to validation report:

```python
{'graph': {
...
  'isolated_nodes': {'has_isolated_nodes': 'True', 'nodes': ['1']},
...}
}
```

and adds extra logging while running the report:

```
2022-08-26 11:36:48,203 - Checking validity of the Network
2022-08-26 11:36:48,203 - Checking validity of the Network graph
...
2022-08-26 11:36:48,204 - This Network has isolated nodes! Consider cleaning it up with `remove_isolated_nodes`
...
```

Bonus: corrects misspelt method `remove_unsused_stops` -> `remove_unused_stops` (the PT equivalent of isolated nodes)

Nb. More PRs in the validation theme coming up soon. I'm leaving updating the notebook: [6.1. Validating Network: MATSim Specific](https://github.com/arup-group/genet/blob/main/notebooks/6.1.%20Validating%20Network:%20MATSim%20Specific.ipynb) to last in the series.

### CODE REVIEW

File: genet/core.py [hunk 1]
```diff
@@ -1839,6 +1839,20 @@ def has_node(self, node_id):
     def has_nodes(self, node_id: list):
         return all([self.has_node(node_id) for node_id in node_id])
 
+    def has_isolated_nodes(self) -> bool:
+        return bool(self.isolated_nodes())
```
Comment (ana-kop) [inline genet/core.py:1843]: Not sure if this is needed as a separate function?

File: genet/core.py [hunk 2]
```diff
@@ -2023,6 +2037,14 @@ def generate_validation_report(self, modes_for_strong_connectivity=None, link_me
             graph_connectivity[mode] = self.check_connectivity_for_mode(mode)
         report['graph'] = {'graph_connectivity': graph_connectivity}
 
+        has_isolated_nodes = self.has_isolated_nodes()
+        report['graph']['isolated_nodes'] = {
+            'has_isolated_nodes': str(has_isolated_nodes),
```
Comment (ana-kop) [inline genet/core.py:2042]: I would maybe get rid of this line and then just keep an `isolated_nodes` key for simplicity, so it just shows `None` if there aren't any? Or maybe at least make it a number of isolated nodes, then have a separate list of them under `nodes`?

**Review by ana-kop — APPROVED** [review]
Looks great, I just added some comments about whether we really need to have a Boolean which says if there are any isolated nodes, I think maybe a number of isolated nodes would make more sense - but up to you :)

File: genet/core.py (hunk 1 above)
Comment (KasiaKoz) [reply genet/core.py:1843]: it's just a convenience method, makes the code more readable later on

File: genet/core.py (hunk 2 above)
Comment (KasiaKoz) [reply genet/core.py:2042]: good idea, i'll make it so:
```python
report['graph']['isolated_nodes'] = {
    "number_of_nodes: 3,
    "nodes" : [A, B, C]
}
```
or 
```python
report['graph']['isolated_nodes'] = {
    "number_of_nodes: 0,
    "nodes" : []
}
```
if there aren't any 👍

**Pull request merged**